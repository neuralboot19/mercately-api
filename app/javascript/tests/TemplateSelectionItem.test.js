import React from 'react';
import { shallow } from 'enzyme';
import TemplateSelectionItem from '../components/WhatsApp/TemplateSelectionItem';

describe('tests over <TemplateSelectionItem/>', () => {
  const setTemplateType = jest.fn();
  const getCleanTemplate = jest.fn();
  const selectTemplate = jest.fn();
  const template = {
    id: 90,
    retailer_id: 241,
    text: "Your * number * has been *.",
    status: "active",
    created_at: "2020-12-03T22:15:59.645Z",
    updated_at: "2020-12-03T22:21:28.093Z",
    template_type: "text",
    gupshup_template_id: "255935d2-5707-4bfd-8185-ac4a55bdf621"
  };

  it('should render on desktop', () => {
    expect.hasAssertions();
    const wrapper = shallow(
      <TemplateSelectionItem
        template={template}
        onMobile={false}
        setTemplateType={setTemplateType}
        getCleanTemplate={getCleanTemplate}
        selectTemplate={selectTemplate}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render on mobile', () => {
    expect.hasAssertions();
    const wrapper = shallow(
      <TemplateSelectionItem
        template={template}
        onMobile
        setTemplateType={setTemplateType}
        getCleanTemplate={getCleanTemplate}
        selectTemplate={selectTemplate}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
