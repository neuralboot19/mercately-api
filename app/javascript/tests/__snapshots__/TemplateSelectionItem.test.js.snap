// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests over <TemplateSelectionItem/> should render on desktop 1`] = `
<div
  className="row"
  key="90"
>
  <div
    className="col-md-10"
  >
    <p>
      [undefined] 
    </p>
  </div>
  <div
    className="col-md-2"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      Seleccionar
    </button>
  </div>
</div>
`;

exports[`tests over <TemplateSelectionItem/> should render on mobile 1`] = `
<div
  className="row"
  key="90"
>
  <div
    className="col-md-10 fs-10"
  >
    <p>
      [undefined] 
    </p>
  </div>
  <div
    className="col-md-2"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      Seleccionar
    </button>
  </div>
</div>
`;
